src/compressed_tensors/transforms/utils.py

```diff
@@ -15,7 +15,27 @@
 import torch
 
 
-__all__ = ["apply_matrix_transform"]
+__all__ = ["apply_matrix_transform", "SingletonMatrixRegistry"]
+
+
+class SingletonMatrixRegistry:
```

Comment:
Is a new class singleton necessary? Since each matrix Transform requires own registry, shouldn't this be implemented on the class itself?

```python3
class Hadamard(Transforms):
    registry: Dict[int, torch.Tensor] = {}

    def __new__(cls, size, empty, transform_name, *args, **kwargs):
        if empty:
            matrix = ...
        else:
            matrix = cls.registry.get(size, torch.Tensor(deterministic_hadamard_matrix(size=self.size)))
        return super().__new__(transform=matrix, transform_name=transform_name)
````